### PR TITLE
Add test for #781

### PR DESCRIPTION
**Pull request by sungam3r**
See #781

### CODE REVIEW

**Comment by Shane32** [comment]
As previously stated, I don't like tests that are dependent on a time factor.  I've found them to be unreliable, as if the azure VM is running slow, then it breaks the test.  Plus, in this particular test, there's nothing to say that the opposite behavior is undesired.

Merge if you like; but I wouldn't.

File: src/GraphQL.Tests/Bugs/Bug781UnobservedTasks.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                    options.ThrowOnUnhandledException = true; // required
+                });
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+
+            GC.Collect(); // GC causes UnobservedTaskException event
+            await Task.Delay(1000); // Wait some time for GC to complete
```
Comment (Shane32) [inline src/GraphQL.Tests/Bugs/Bug781UnobservedTasks.cs:40]:
```suggestion
            GC.Collect(); // call finalizer of Task
            await Task.Delay(1000); // Wait some time for GC to complete
            GC.Collect(); // Task is collected and GC causes UnobservedTaskException event
            await Task.Delay(1000); // Wait some time for GC to complete
```

Comment (Shane32) [reply src/GraphQL.Tests/Bugs/Bug781UnobservedTasks.cs:40]: This fixes the test, at least on my machine.  Why?  I don't know; I'd just be guessing.  Presumably you had the code working before.  Another example of the test being unreliable.

**Comment by Shane32** [comment]
`[Fact(Skip)]` 👍

**Comment by sungam3r** [comment]
Indeed the results were unstable.

**Pull request merged**